Assets/MixedRealityToolkit/Interfaces/Registrars/IMixedRealityServiceRegistrar.cs

```diff
@@ -0,0 +1,143 @@
+
+        /// <summary>
+        /// Registers a service of the specified type.
+        /// </summary>
+        /// <typeparam name="T">The interface type of the service to be registered (ex: IMixedRealityBoundarySystem).</typeparam>
+        /// <param name="concreteType">The concrete type to instantiate.</param>
+        /// <param name="supportedPlatforms">The platform(s) on which the service is supported.</param>
+        /// <param name="args">Optional arguments used when instantiating the concrete type.</param>
+        /// <returns>True if the service was successfully registered, false otherwise.</returns>
+        bool RegisterService<T>(
+            Type concreteType,
+            SupportedPlatforms supportedPlatforms = (SupportedPlatforms)(-1),
+            params object[] args) where T : IMixedRealityService;
+
+        /// <summary>
+        /// Unregisters a service of the specified type.
+        /// </summary>
+        /// <typeparam name="T">The interface type of the service to be unregistered (ex: IMixedRealityBoundarySystem).
+        /// <param name="name">The name of the service to unregister.</param>
+        /// <returns>True if the service was successfully unregistered, false otherwise.</returns>
+        /// <remarks>If the name argument is not especified, the first instance will be unregistered</remarks>
+        bool UnregisterService<T>(string name = null) where T : IMixedRealityService;
+
+        /// <summary>
+        /// Unregisters a service.
+        /// </summary>
+        /// <typeparam name="T">The interface type of the service to be unregistered (ex: IMixedRealityBoundarySystem).
+        /// <param name="service">The specific service instance to unregister.</param>
+        /// <returns>True if the service was successfully unregistered, false otherwise.</returns>
+        bool UnregisterService<T>(T serviceInstance) where T : IMixedRealityService;
+
+        /// <summary>
+        /// Checks to see if a service of the specified type has been registered.
+        /// </summary>
+        /// <typeparam name="T">The interface type of the service (ex: IMixedRealityBoundarySystem).
+        /// <param name="name">The name of the service.</param>
+        /// <returns>True if the service is registered, false otherwise.</returns>
+        bool IsServiceRegistered<T>(string name = null) where T : IMixedRealityService;
+
+        /// <summary>
+        /// Gets the instance of the registered service.
+        /// </summary>
+        /// <typeparam name="T">The interface type of the service (ex: IMixedRealityBoundarySystem).
+        /// <param name="name">The name of the service.</param>
+        /// <param name="showLogs">Indicates whether or not diagnostic logging should be performed in case of an error</param>
+        /// <returns>The registered service instance as the requested type.</returns>
+        T GetService<T>(string name = null, bool showLogs = true) where T : IMixedRealityService;
```

Comment:
for the showLogs parameter, could you provide a little more background in how this would be used as well? i.e. usually I would imagine that logging is a general system thing (i.e. Logging.enabled) rather than having logging enabled on a per function parameter basis.

i.e. if we wanted to enable logs on other functions, would this mean we'd have to update them to also have a showLogs parameter?

Full transparency, I don't have super deep background in the way things work already, so I'm just more wondering out loud how this works + what patterns we've adopted.

Reply:
There was previously an issue with customers seeing _lots_ of debug output if a service could not be acquired (ex: was not loaded). The showLogs param was an attempt to reduce that output to one message when there was a legitimate issue finding the serivce.

Reply:
Got it - I see the existing code which had this.

I think for future I would probably try to push us to augment our logging to have different criticalities (i.e. info, warn, error) and do filtering based on that so that we can avoid littering the log with more noisy stuff but still switch it on in cases where we want to debug further.

But yeah, this is a separate thing from this PR :)